### PR TITLE
Removed signed in user from the list of users for sharing

### PR DESCRIPTION
**Pull request by fesave**
## Related Issues
App: #1419 

- [x] Added changelog files for the fixed issues in folder changelog/unreleased. More info [here](https://github.com/owncloud/android/tree/master/changelog#create-changelog-items)
_____

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt [hunk 1]
```diff
@@ -180,6 +179,9 @@ class UsersAndGroupsSearchProvider : ContentProvider() {
             try {
                 while (namesIt.hasNext()) {
                     item = namesIt.next()
+                    if (item.label == AccountUtils.getUsernameOfAccount(account.name) && item.shareType == ShareType.USER) {
+                        item = namesIt.next()
```
Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt:183]: Please, don't do this.

If we use an iterator to iterate in a while loop, the condition that checks if we need to exit the loop is `hasNext()`. So, we use`next()` only when we are sure it `hasNext()`. If we want to proceed with the next step, [we use `continue`](https://kotlinlang.org/docs/returns.html)

Comment (abelgardep) [inline owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt:183]: Please, don't do this.

If we use an iterator to iterate in a while loop, the condition that checks if we need to exit the loop is `hasNext()`. So, we use`next()` only when we are sure it `hasNext()`. If we want to proceed with the next step, [we use `continue`](https://kotlinlang.org/docs/returns.html)

**Review by abelgardep — CHANGES_REQUESTED** [review]
Some changes here @fesave

File: changelog/unreleased/3643 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Bugfix: Fixed a bug in the list of users to be shared
```
[the remark this answers is not in the document]
Comment (fesave) [reply changelog/unreleased/3643:1]: Changed

File: owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt [hunk 3]
```diff
@@ -180,6 +179,9 @@ class UsersAndGroupsSearchProvider : ContentProvider() {
             try {
                 while (namesIt.hasNext()) {
                     item = namesIt.next()
+                    if (item.label == AccountUtils.getUsernameOfAccount(account.name) && item.shareType == ShareType.USER) {
```
[the remark this answers is not in the document]
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt:182]: Done

File: owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt (hunk 1 above)
Comment (fesave) [reply owncloudApp/src/main/java/com/owncloud/android/presentation/providers/sharing/UsersAndGroupsSearchProvider.kt:183]: Understood! Thank u a lot.

**Comment by jesmrec** [comment]
- If username == displayname -> ✅   . Username is not in the list, not selectable

- If username != displayname  -> ❌   . It's selectable. For example, this user:

Screenshot 2022-05-04 at 13 32 55

searching:

Screenshot 2022-05-04 at 13 32 45

it should not appear either.

Could you take a look, @fesave ?

**Comment by fesave** [comment]
> * If username == displayname -> ✅   . Username is not in the list, not selectable
> * If username != displayname  -> ❌   . It's selectable. For example, this user:
> 
> Screenshot 2022-05-04 at 13 32 55
> 
> searching:
> 
> Screenshot 2022-05-04 at 13 32 45
> 
> it should not appear either.
> 
> Could you take a look, @fesave ?

Good catch! Fixed :)

**Comment by jesmrec** [comment]
Problem with the display name is fixed. So, everything done.

Approved

**Pull request merged**